Accept optional predicted usage authorize and oauth_authorize calls

davidor opened the pull request:
Closes #28

unleashed commented:
Functionality wise this should be ok. But can't stop thinking we could use the opportunity for some refactoring.

mikz commented on lib/3scale/client.rb:
```diff
@@ -285,6 +287,13 @@ def encode_transactions(transactions)
+      usage.map do |metric, value|
+        escaped_metric = CGI.escape "[usage][#{metric}]"
+        "#{escaped_metric}=#{CGI.escape(value.to_s)}"
+      end.join('&')
```
Imo `URI.encode_www_form` could be help here. 

```ruby
URI.encode_www_form(usage.map { |metric, value| ["[usage][#{metric}]", value ] })
```

davidor replied:
I just extracted the method as it was. But yes, while we are at it I can change it. Your solution looks cleaner.

mikz commented:
Yeah. Maybe a tiny bit of refactoring would not hurt. But just on the touched lines :)

 Otherwise 👍

mayorova commented:
It was surprising for me that `app_id=foo&[usage][metric1]=1&[usage][metric2]=2` apparently works the same way that `app_id=foo&usage[metric1]=1&usage[metric2]=2`, but OK :)

davidor commented:
@mikz I addressed your comment and rebased.

@mayorova Yes, the original code was using the `[usage]` version, I didn't change that.

@unleashed I thought the same. There are some opportunities for refactoring, but I think it's better to tackle that in a separate PR when we have the time, as @mikz suggested, because any refactoring that we could make would probably affect many other things, not just the ones changed in this PR.
